event-management-frontend/src/components/userMapEvents/UserMapEventsSmart.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react';
+import { EventCard } from '../../model/userHome/EventCard';
+import { fetchUserUpcomingEvents } from '../../actions/UserHomePageActions';
+import { AppState } from '../../store/store';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
+import { CircularProgress } from '@material-ui/core';
+import { EventProps } from '../userHomePage/eventsSection/EventsSectionSmart';
+import { LocationType } from '../../model/LocationType';
+import UserMapEventsDumb from './UserMapEventsDumb';
+
+interface UserHomePageProps {
+  upcomingEvents: EventProps;
+  fetchUserUpcomingEvents: (page: number, limit: number) => void;
+}
+
+export interface EventCardsWithLoction {
+  eventCards: EventCard[];
+  location: LocationType;
+}
+
+const UserMapEventsSmart = (props: UserHomePageProps) => {
+  const [eventCardsWithLocation, setEventCardsWithLocation] = useState<EventCardsWithLoction[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  var result: EventCardsWithLoction[] = [];
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    fetchUserUpcomingEvents(0, 30);
```

Comment:
If this is pagination, I would define a locally-scoped global in which these would set as default values.